airflow/providers/sftp/hooks/sftp.py

```diff
@@ -120,12 +120,11 @@ def get_conn(self) -> pysftp.Connection:
 
     def close_conn(self) -> None:
         """
-        Closes the connection. An error will occur if the
-        connection wasnt ever opened.
+        Closes the connection
         """
-        conn = self.conn
-        conn.close()  # type: ignore
-        self.conn = None
+        if self.conn is not None:
+            self.conn.close()
+            self.conn = None
```

Comment:
And I wander, could we just do 
```py
if not self.conn:
    self.conn.close()
```
If we already check if conn is null or not?

Reply:
I think when we initialize SFTPHook, self.conn is None.
This may cause an error in two case
1. call `close_conn` without a connection
2. call `close_conn` more than once after connection

it's not necessary to cause an error in this case

Reply:
Yeah. Setting it to none is good.